File: frontend/src/features/link/types/link.ts
export interface Category {
  id: number;
  name: string;
}

export interface LinkRequestDto {
  title?: string;
  url: string;
}

export interface LinkResponseDto {
  id: number;
  title: string;
  url: string;
  thumbnailImageUrl: string;
  price: string;
  previewStatus: PreviewStatus;
}

export interface LinkPreviewDto {
  title: string;
  thumbnailImageUrl: string;
  price: string;
}

export enum PreviewStatus {
  PENDING = "PENDING",
  COMPLETED = "COMPLETED",
  FAILED = "FAILED",
}

File: backend/src/main/java/com/project/bearlink/domain/link/entity/Link.java
package com.project.bearlink.domain.link.entity;

import com.project.bearlink.domain.category.entity.Category;
import com.project.bearlink.domain.user.user.entity.User;
import com.project.bearlink.global.jpa.BaseEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Table(name = "link")
public class Link extends BaseEntity {

    private String title;
    private String url;
    private String thumbnailImageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;

    private String price;

    @Enumerated(EnumType.STRING)
    private PreviewStatus previewStatus;
}
